aiidalab_widgets_base/structures.py

```diff
@@ -1,88 +1,138 @@
 from __future__ import print_function
 from __future__ import absolute_import
 
+import os
 import ase.io
 import ipywidgets as ipw
 from fileupload import FileUploadWidget
 import tempfile
 import nglview
 from six.moves import zip
+from traitlets import Bool
 
 
 class StructureUploadWidget(ipw.VBox):
-
+    has_structure = Bool(False)
     DATA_FORMATS = ('StructureData', 'CifData')
-
-    def __init__(self, text="Upload Structure", node_class=None, **kwargs):
+    def __init__(self, text="Upload Structure", storable=True, node_class=None, examples=[], **kwargs):
         """ Upload a structure and store it in AiiDA database.
 
```

Comment:
perhaps add an explanation here of how this class should be used (`has_structure`, `structure_node`)

Reply:
Ok, I will add